### PR TITLE
Feature/authorization rules

### PR DESCRIPTION
**Pull request by Alejuda**
Hello there! in this branch, we worked in collaboration with my coding partner @EmiLoKiJu and did the following:

- Install CanCanCan gem.
- Add role column to the users table
- A user can delete a post if it is theirs or if they have an admin role
  -  Implement the post deleting functionality. Only authorized users can see the "Delete" button
- A user can delete a comment if it is theirs or if they have an admin role 
  - Implement the comment deleting functionality. Only authorized users can see the "Delete comment" button

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -23,4 +23,14 @@ def create
       end
     end
   end
+
+  def destroy
+    comment = Comment.find(params[:id])
```
Comment (KDlamini) [inline app/controllers/comments_controller.rb:28]: After a comment is deleted, I noticed that the `comments_counter` is not updated on the page. The post still indicated that there was a comment even when there were no comments on the post.

- Kindly look into it and make sure that everything is consistent on the application including the counters.

File: app/controllers/posts_controller.rb
Comment (KDlamini) [inline app/controllers/posts_controller.rb:1]: Kindly make sure that the application uses **CanCanCan**'s `load_and_authorize_resource` for Post authorization in `Post Controller`.

```ruby
  class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
```

**Review by KDlamini — CHANGES_REQUESTED** [review]
### Hi team @Alejuda, @EmiLoKiJu 🙋‍♀️,  

Excellent work! 💯 
You have done a great job adding authorization rules for the blog app. 🎊 🥳 
There are some issues that you still need to work on to improve your project according to the requirements. You are almost there! 💪🏾 

## Highlights 🌞

✔️  Good PR title and description
✔️  Well-documented and professional README 
✔️  The `role` column is created and exists in the user's table
✔️  There's a CanCanCan `ability` class that defines the set of rules for user post interactions
✔️  Users can only delete Posts and Comments that they own

## Required Changes ♻️
 
**Kindly look through the comments below 👇 for required changes**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb
Comment (Alejuda) [reply app/controllers/posts_controller.rb:1]: Hello @KDlamini thank you for the review! it was really helpful. I already implemented the required changes and the app is consistent on the application.

**Review by Tufahel — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

Hi @Alejuda & @EmiLoKiJu  :wave: 
 
Well done on implementing the required changes. 👍 
Your project is complete now!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 👏🏻



:heavy_check_mark:  Fixed comment counting issue.

## Optional suggestions 🤷‍♂️

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.
- You can mail me at [Mail](mailto:tufahel96@gmail.com) or ask any question at Slack, `Name: Tufahel`

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**